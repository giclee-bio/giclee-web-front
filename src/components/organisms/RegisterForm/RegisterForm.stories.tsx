import { Meta, StoryObj } from '@storybook/react'

import RegisterForm from './RegisterForm'

const meta: Meta<typeof RegisterForm> = {}

export default meta
type Story = StoryObj<typeof RegisterForm>

export const Index: Story = {
  args: {},
}
