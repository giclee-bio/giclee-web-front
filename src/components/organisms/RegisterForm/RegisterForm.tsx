'use client'
import { useSession } from 'next-auth/react'

import Form from '@/components/molecules/Form'

import type { Props as ButtonProps } from '@/components/atoms/Button'
import type { FormField } from '@/components/molecules/Form'

const buttonProps: ButtonProps = {
  children: '登録する',
  type: 'primary',
}

const formFields: FormField[] = [
  {
    label: 'ユーザー名',
    name: 'username',
    require: true,
    tag: 'input',
    type: 'text',
  },
]

const RegisterForm = () => {
  const { data: session } = useSession()
  formFields[0].defaultValue = session?.user?.name || undefined
  return (
    <section className='flex h-96 w-80 flex-col items-center justify-between rounded bg-stone-800 p-8'>
      <h1 className='text-2xl font-bold'>ユーザー登録</h1>
      <p>ユーザー名は後から変更することができません。</p>
      <Form buttonProps={buttonProps} formFields={formFields} />
    </section>
  )
}

export default RegisterForm
