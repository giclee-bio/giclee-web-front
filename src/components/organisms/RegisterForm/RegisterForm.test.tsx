import { render } from '@testing-library/react'

import RegisterForm from './RegisterForm'

describe('RegisterForm', () => {
  it('snapshot', async () => {
    const { asFragment } = render(<RegisterForm />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
