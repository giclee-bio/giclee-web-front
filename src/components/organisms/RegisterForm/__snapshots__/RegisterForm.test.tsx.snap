// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm snapshot 1`] = `
<DocumentFragment>
  <section
    class="flex h-96 w-80 flex-col items-center justify-between rounded bg-stone-800 p-8"
  >
    <h1
      class="text-2xl font-bold"
    >
      ユーザー登録
    </h1>
    <p>
      ユーザー名は後から変更することができません。
    </p>
    <form
      class="w-64"
    >
      <div
        class="mt-4 first:mt-0"
      >
        <div
          class="flex justify-between text-sm"
        >
          <label
            class="font-bold"
            for="radix-:r0:"
          >
            ユーザー名
          </label>
        </div>
        <input
          class="mt-2 bg-stone-50 h-8 w-full rounded px-3 text-black"
          id="radix-:r0:"
          name="username"
          required=""
          title=""
          type="text"
          value=""
        />
      </div>
      <button
        class="mt-9 block mx-auto h-10 rounded text-center font-bold leading-10 w-64 bg-primary text-black"
      >
        登録する
      </button>
    </form>
  </section>
</DocumentFragment>
`;
